New exporters and small fixes

viktorradnai opened the pull request:
- New exporters: mysql and wmi (for Windows).
- Each resource sets an attribute so a Prometheus server can discover the exporters using knife search.

The wmi_exporter is an odd one as it requires Windows. I have merged it but it may be better placed in a separate `prometheus_windows_exporters` cookbook.

viktorradnai commented:
@dragonsmith Sorry to bug you but have you got a rough idea on whether this can be merged and when? We're currently working on adding more exporters so there might be another PR or an update to this one in a week or two.

dragonsmith commented:
@viktorradnai let me see into it today, sorry for being unresponsive.

viktorradnai commented:
@dragonsmith Thanks for this. I'm actually going to back out the blackbox exporter stuff from this PR and create a new one that can be applied separately. I'm just having "fun" managing commits from different sources and squashing commits... I'll let you know as soon as I'm done.

dragonsmith commented:
Uh, sorry, missed the attribute in a haste.

dragonsmith commented:
wmi_exporter attribute for knife search was missing though...

Anyway, 0.8.1 is being tested on TravisCI right now.

viktorradnai commented:
@dragonsmith if you have a way to back out this PR then please do so. I already have a version where everything is fixed, all the commits are squashed and I'm just adding documentation that @flamingspaz missed. I accidentally pushed @flamingspaz's original PR here rather than the version I've already squashed and fixed. That branch will most likely conflict this one.

dragonsmith commented:
@viktorradnai the tests finished and 0.8.1 was released a few minutes ago... https://supermarket.chef.io/cookbooks/prometheus_exporters/versions/0.8.1

Can you open a new PR if there is more to add, please?

viktorradnai commented:
Thanks. Opened #10
